container-api/src/main/java/org/jboss/forge/furnace/repositories/AddonStateRepository.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2012 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Eclipse Public License version 1.0, available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.jboss.forge.furnace.repositories;
+
+import org.jboss.forge.furnace.addons.Addon;
+import org.jboss.forge.furnace.addons.AddonId;
+import org.jboss.forge.furnace.versions.Version;
+
+import java.util.List;
+
+/**
+ * Used to perform {@link Addon} registration operations.
+ * 
+ * @author <a href="mailto:lincolnbaxter@gmail.com">Lincoln Baxter, III</a>
+ * @author <a href="mailto:koen.aers@gmail.com">Koen Aers</a>
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+public interface AddonStateRepository
```

Comment:
Yes, you are right that the `AddonRepository` interface definition looks weird. I slighlty forgot that `AddonRepository` was extending those interfaces.

I now see that it is an implementation detail for the `AddonRepositoryImpl` class. Perhaps we should leave the API untouched and keep the implementation changes only? 

Sorry, I haven't realized this before seeing this change.